test/GameLogic/e2e/ActionRecording.test.ts

```diff
@@ -0,0 +1,28 @@
+import 'mocha';
```

Comment:
Looks good, can you also add unit tests for `ActionSequence` itself, checking all the cases you can think of that make sense?

Reply:
Added some ActionSequence tests